bin/jobsub_cmd

```diff
@@ -17,37 +17,65 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-schedd=""
-case "x$*" in
-*@*) schedd=`echo "$*" | sed -e 's/.*@//' -e 's/ .*//'`
-     args=`echo "$*" | sed -e 's/@[^ ]*//'`
-     ;;
-*)  args="$*"
-    ;;
-esac
-if [ x$schedd != x ]
-then
-    args="-name $schedd $args"
-fi
-
-# Check group and pull it into environment
-case "x$args" in
-*--group*)
-    group=`echo "$args" | sed -e 's/.*--group[=( )+]//' -e 's/ .*//'`
-    args=`echo "$args*" | sed -e "s/--group[=( )+]$group//"`
-    ;;
-*-G*)
-    group=`echo "$args" | sed -e 's/.*-G[( )+]//' -e 's/ .*//'`
-    args=`echo "$args" | sed -e "s/-G[( )+]$group//"`
-    ;;
-esac
-if [ x$group != x ]
-then
-    export GROUP=$group
-fi
-
-export X509_USER_PROXY=`fake_ifdh getProxy`
-export BEARER_TOKEN_FILE=`fake_ifdh getToken`
-
-cmd=`basename $0 | sed -e 's;^[a-z]*_;condor_;'`
-exec /usr/bin/$cmd -name $schedd $args
+import argparse
+import os
+import subprocess
+import sys
+
+class StoreGroupinEnvironment(argparse.Action):
+    """Action to store the given group in the GROUP environment variable"""
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        os.environ["GROUP"] = values
+        setattr(namespace, self.dest, values)
+
+
+def backquote(cmd):
+    sp = subprocess.Popen(cmd, shell=True, encoding='UTF-8', stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    res, stderr = sp.communicate()
+    return res
+
+def main():
+    parser = argparse.ArgumentParser(description = 'Condor command wrapper')
+    parser.add_argument(
+        "-G",
+        "--group",
+        help="Group/Experiment/Subgroup for priorities and accounting",
+        action=StoreGroupinEnvironment,
+    )
+    parser.add_argument('passthru', nargs='+', action='append' )
+
+
+    arglist=parser.parse_args()
+
+    # make list of arguments to pass to condor command:
+    # - the passthru arguments from above, except if we have
+    #   any 234@schedd style arguments, pick out the schedd and
+    #   keep the 234, and pass --name schedd as well
+    execargs = []
+    schedd = None
+    for i in arglist.passthru[0]:
+        pos = i.find('@')
+        if pos >= 0:
+            schedd = i[pos+1:]
+            i = i[0:pos]
+        execargs.append(i)
+
+    if schedd:
+        execargs.insert(0, schedd)
+        execargs.insert(0, "-name")
+
+    # also make sure we have suitable credentials...
+    os.environ['X509_USER_PROXY'] = backquote('fake_ifdh getProxy')
```

Comment:
So we're spawning a sub-process to call a python script? I assume this is to keep the python script as close to the shell as possible but I'd really rather see it done "right" now (which in my mind is making the token and proxy functions into a library) than hoping/planning to do that later.

Reply:
Okay, so now it imports fake_ifdh and makes the calls.